Gems/Atom/RPI/Code/Source/RPI.Public/Pass/RasterPass.cpp

```diff
@@ -99,6 +99,116 @@ namespace AZ
             }
         }
 
+        bool RasterPass::BuildSubpassLayout(
+            RHI::RenderAttachmentLayoutBuilder::SubpassAttachmentLayoutBuilder& subpassLayoutBuilder,
+            uint32_t subpassIndex)
+        {
+            if (!m_flags.m_mergeChildrenAsSubpasses)
+            {
+                AZ_Error("RasterPass", false, "%s can only be called on RasterPasses that are supposed to be merged as subpasses.", __FUNCTION__);
+                AZ_Assert(false, "%s can only be called on RasterPasses that are supposed to be merged as subpasses.", __FUNCTION__);
```

Comment:
Why both an error and an assert? Can we do with just one?

Reply:
I left both because in Release builds the AZ_Asserts are removed. This error is clearly a programmer's error so in debug and profile it should assert. At least in Release there will be an explanation.